src/devices/tuya.ts

```diff
@@ -7712,6 +7753,75 @@ const definitions: Definition[] = [
             ],
         },
     },
+    {
+        fingerprint: tuya.fingerprint('TS0601', ['_TZE204_dvosyycn']),
+        model: 'TS0601',
+        vendor: 'TuYa',
+        description: 'ZXYH 8IN/8OUT',
+        extend: [],
+        fromZigbee: [tuya.fz.datapoints, fzLocal.TZE204_dvosyycn],
+        toZigbee: [tuya.tz.datapoints, tzLocal.TZE204_dvosyycn],
```

Comment:
Please don't introduce the tzLocal/fzLocal converters, use the `tuya.fz.datapoints`/`tuya.tz.datapoints` instead.

Reply:
I'm really sorry, but I guess my coding skills are not relly good enough... As discussed I used to have everything in one file, including the fz/tzLocal converters..

I think I understand how to avoid the `tzLocal` and just use the `tuya.tz.datapoints` as is (as it seems to do the same), but for the `fzLocal` I'm not clear on how this should work, as there need to be some calculations done, old variables read and new variables set, when the values arrive from the device...

Reply:
You can add your own value converter instead, see e.g. this example:

https://github.com/Koenkk/zigbee-herdsman-converters/blob/eedb8a692b4e5ef1ca9f05bd44686e61857e00ff/src/lib/tuya.ts#L896

Reply:
I'm really sorry, but I don't succeed. I don't know what I'm doing wrong. I'm struggling with the format.

Same for the other 2 devices I did local templates which work fine in my environment (see issue tracker), but also have the fz_local and tz_local functions within the file...

Reply:
I don't understand why this converter is needed, what breaks when removing it?

Reply:
Not sure if I understand your question right, probably my english's not good enough... This converter is for changing the function of the input pins. With this device you have to select first witch IO you want to change (value1) and then to what fuction you want to change it (value2).

By default the input's are coupled to the output, meaning when you set I1 to High, O1 also goes to high. But this is changeable by setting value1 to 1 and then value2 to the desired function (i.e. decouple, blink, etc.). And I didn't find in any existing device this function/possibility.

Hope that clarifies...

Reply:
I've created an external definition with a custom value converter which should fix the issue: https://gist.github.com/Koenkk/26dd2afc0a5300b59de9c5a9e64dd82d.

Note that this required tomorrows new z2m release, so let me know if it works with that.

Reply:
PS: I've made 3 other definition/js files for devices I use which work fine. But unfortunately I still don't understand how I get from the .js to the .ts format.... I'd love to help and js is somehow understandable, but the rest is not my world at all... really sorry... 
if someone wnats to take my js files and include them in the project, please do so...
thanks for your time and efforts, z2m is a really great project!

Reply:
If you provide the js definitions, I can integrate them.